src/bin/cargo-flamegraph.rs

```diff
@@ -316,11 +316,53 @@ fn workload(
     result
 }
 
+fn find_unique_bin_target() -> String {
+    let metadata = MetadataCommand::new()
+        .no_deps()
+        .exec()
+        .expect("failed to access crate metadata");
+
+    let bin_targets: Vec<String> = metadata
+        .packages
+        .into_iter()
+        .flat_map(|p| p.targets.into_iter())
+        .filter_map(|t| {
+            t.kind.contains(&"bin".into()).then(|| t.name)
+        })
+        .collect();
+
+    match bin_targets.as_slice() {
+        [target] => {
+            eprintln!(
+                "automatically selected {} as it is the only binary target",
+                target
+            );
+            target.clone()
```

Comment:
I think we can return `bin_targets[0]` here instead of cloning?

Reply:
No:
```
Error[E0507]: cannot move out of index of `Vec<BinaryTarget>`
   --> src/bin/cargo-flamegraph.rs:367:13
    |
367 |             bin_targets[0]
    |             ^^^^^^^^^^^^^^ move occurs because value has type `BinaryTarget`, which does not implement the `Copy` trait
```

I managed to make it work but I'm not convinced that's better:
```rust
match bin_targets.as_slice() {
    [_] => bin_targets.into_iter().next().unwrap(),
...
```

While developing I also tried to destructure the `Vec` itself but that also leads to borrow checker issues and unsized locals for the catch-all arm.

Reply:
Ok, just had another look into `Vec` docs. `bin_targets.remove(0);`works as well if I make `bin_targets` mutable. I think I will go with that.

Reply:
Yup, that sounds good!